meta-mentor-staging/classes/scancode.bbclass

```diff
@@ -0,0 +1,24 @@
+# We use scancode utlity for extacting licence information.
+# scancode itself is an OSS Utlitity.
+# For more informaiton https://github.com/nexB/scancode-toolkit
+# scancode is used from HOSTTOOLS.
+
+SCANCODE_FORMAT ?= "html-app"
+EXT = "${@'html' if d.getVar('SCANCODE_FORMAT', True) == 'html-app' else 'json'}"
+
+do_scancode() {
+	mkdir -p ${DEPLOY_DIR_IMAGE}/scancode
+	if [ -d "${S}" ]; then
+		scancode ${S} --format  ${SCANCODE_FORMAT} ${DEPLOY_DIR_IMAGE}/scancode/${PN}.${EXT}
```

Comment:
The convention is to no longer write directly to DEPLOY_DIR_IMAGE, but instead allow the sstate state management code to keep track of what it's installed and uninstall to/from that location. I'd suggest doing the same here, following the same pattern as that used by deploy.bbclass.

Reply:
Upon further consideration, I'd also suggest moving this to its own subdirectory of DEPLOY_DIR, rather than cluttering up DEPLOY_DIR_IMAGE with content that isn't directly bound to an image.